test/lint/commit-script-check.sh

```diff
@@ -20,23 +20,23 @@ fi
 RET=0
 PREV_BRANCH=`git name-rev --name-only HEAD`
 PREV_HEAD=`git rev-parse HEAD`
-for i in `git rev-list --reverse $1`; do
-    if git rev-list -n 1 --pretty="%s" $i | grep -q "^scripted-diff:"; then
-        git checkout --quiet $i^ || exit
-        SCRIPT="`git rev-list --format=%b -n1 $i | sed '/^-BEGIN VERIFY SCRIPT-$/,/^-END VERIFY SCRIPT-$/{//!b};d'`"
+for commit in `git rev-list --reverse $1`; do
+    if git rev-list -n 1 --pretty="%s" $commit | grep -q "^scripted-diff:"; then
+        git checkout --quiet $commit^ || exit
+        SCRIPT="`git rev-list --format=%b -n1 $commit | sed '/^-BEGIN VERIFY SCRIPT-$/,/^-END VERIFY SCRIPT-$/{//!b};d'`"
         if test "x$SCRIPT" = "x"; then
-            echo "Error: missing script for: $i"
+            echo "Error: missing script for: $commit"
             echo "Failed"
             RET=1
         else
-            echo "Running script for: $i"
+            echo "Running script for: $commit"
             echo "$SCRIPT"
-            eval "$SCRIPT"
-            git --no-pager diff --exit-code $i && echo "OK" || (echo "Failed"; false) || RET=1
+            (eval "$SCRIPT")
```

Comment:
Would it make sense to use `(set -e; eval "$SCRIPT") || RET=1 && echo "scripted-diff execution failed."`?

That would allow us to catch execution failures within the scripted-diff run.

It probably shouldn't be done as part of this PR as that would change behaviour.

Reply:
Not sure about `set -e`, as I've found that people are not used to its behaviour and there are legitimate uses of not failing immediately upon the failure of one command. I think that as long as the diff produces the expected result, we should be good.

What I _would_ like to have is `set -x`, which doesn't change behaviour but allows us to see line by line execution fully expanded (in case it's a complex script they'd like to debug).

Reply:
`set -x` would be nice :-)